#!/usr/bin/env python3
PACKAGE = "strategy"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#------------------------------------------------------------------------------------
# Game Start
#------------------------------------------------------------------------------------
gen.add("GAME_START",    bool_t,    0, "A Boolen parameter",  False)

#------------------------------------------------------------------------------------
# Robot mode
#------------------------------------------------------------------------------------
mode_enum = gen.enum([ gen.const("IDLE",         str_t, "idle",         "robot is at ide mode"),
                       gen.const("SETTING",      str_t, "Setting",      "Setting the robot"),
                       gen.const("NAVIGATING",   str_t, "Navigating",   "the robot start to navagate"),
                       gen.const("TEST",  str_t, "test",  "test the robot route"), 
                      #  gen.const("QGA",  str_t, "QGA",  "calculate the robot route"),
                       gen.const("SERVICE",  str_t, "Service",  "Servie the customer")], "An enum to set mode")
gen.add("ROBOT_MODE", str_t, 0, "Select Game mode", "idle", edit_method = mode_enum)



#------------------------------------------------------------------------------------
# item_list
#------------------------------------------------------------------------------------
item_enum = gen.enum([ gen.const("INITIAL",    str_t, "initial",         ""),
                       gen.const("SHELF",   str_t, "shelf",        ""),
                      #  gen.const("item2",   str_t, "item2",        ""), 
                      #  gen.const("item3",   str_t, "item3",        ""),
                       gen.const("TABLE1",   str_t, "Table1",        ""),
                       gen.const("TABLE2",   str_t, "Table2",        "")], "An item enum")
gen.add("ITEM", str_t, 0, "Select the item u want to go", "initial", edit_method = item_enum)

#------------------------------------------------------------------------------------
# setting option
#------------------------------------------------------------------------------------
gen.add("LOC_GET",    bool_t,    0, "A Boolen parameter",  False)
gen.add("LOC_RESET",     bool_t,    0, "A Boolen parameter",  False)
#------------------------------------------------------------------------------------
# nav_option
#------------------------------------------------------------------------------------
nav_mode_enum = gen.enum([  gen.const("test",        str_t, "test",             ""),
                       gen.const("directory",         str_t, "directory",         "")], "An naav mode enum")
gen.add("NAV_MODE", str_t, 0, "Select the nav mode", "directory", edit_method = nav_mode_enum)
gen.add("NAV_START",    bool_t,    0, "A Boolen parameter",  False)
gen.add("NAV_STOP",    bool_t,    0, "A Boolen parameter",  False)
gen.add("YAML_SAVE",    bool_t,    0, "A Boolen parameter",  False)
gen.add("YAML_LOAD",    bool_t,    0, "A Boolen parameter",  False)
exit(gen.generate(PACKAGE, "strategy", "Robot"))






# gen.add("game_start",    int_t,    0, "An Integer parameter", 50,  0, 100)
# gen.add("double_param", double_t, 0, "A double parameter",    .5, 0,   1)
# gen.add("str_param",    str_t,    0, "A string parameter",  "Hello World")
# gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)



